packages/riipen-ui/src/components/Menu.test.jsx

```diff
@@ -0,0 +1,333 @@
+      );
+
+      const wrapper = mount(
+        <Menu anchorEl={anchorEl} classes={classes}>
+          <MenuItem />
+        </Menu>
+      );
+
+      expect(
+        wrapper
+          .find("MenuList")
+          .childAt(0)
+          .hasClass(classes[0])
+      ).toEqual(true);
+    });
+
+    it("appends higher order values to classes prop with withClasses decorator", () => {
+      const classes = ["riipen", "riipen-menu"];
+
+      const wrapper = mount(<Menu />);
+
+      expect(
+        wrapper
+          .find("Menu")
+          .props()
+          .classes.sort()
+      ).toEqual(classes.sort());
+    });
+  });
+
+  describe("color prop", () => {
+    it("passes color prop to correct child", () => {
+      const color = "secondary";
+      const anchorEl = ReactTestUtils.renderIntoDocument(
+        <div>
+          <AppBar />
+        </div>
+      );
+
+      const wrapper = mount(
+        <Menu anchorEl={anchorEl} color={color}>
+          <MenuItem />
+        </Menu>
+      );
+
+      expect(wrapper.find("MenuList").props().color).toEqual(color);
+    });
+
+    it("gives an error when given an invalid color", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+
+      mount(<Menu color="blue" />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("contentPosition prop", () => {
+    it("passes contentPosition prop to correct child", () => {
+      const contentPosition = { horizontal: "left", vertical: "bottom" };
+
+      const wrapper = mount(<Menu contentPosition={contentPosition} />);
+
+      expect(wrapper.find("Popover").props().contentPosition).toEqual(
+        contentPosition
+      );
+    });
+  });
+
+  describe("fullWidth prop", () => {
+    it("passes fullWidth prop to correct child", () => {
+      const fullWidth = true;
+
+      const wrapper = mount(<Menu fullWidth={fullWidth} />);
+
+      expect(wrapper.find("Popover").props().fullWidth).toEqual(fullWidth);
+    });
+  });
+
+  describe("isOpen prop", () => {
+    it("passes isOpen prop to correct child", () => {
+      const isOpen = true;
+
+      const wrapper = mount(<Menu isOpen={isOpen} />);
+
+      expect(wrapper.find("Popover").props().isOpen).toEqual(isOpen);
+    });
+  });
+
+  describe("keepOnScreen prop", () => {
+    it("passes keepOnScreen prop to correct child", () => {
+      const keepOnScreen = true;
+
+      const wrapper = mount(<Menu keepOnScreen={keepOnScreen} />);
+
+      expect(wrapper.find("Popover").props().keepOnScreen).toEqual(
+        keepOnScreen
+      );
+    });
+  });
+
+  describe("marginThreshold prop", () => {
+    it("passes marginThreshold prop to correct child", () => {
+      const marginThreshold = 2;
+
+      const wrapper = mount(<Menu marginThreshold={marginThreshold} />);
+
+      expect(wrapper.find("Popover").props().marginThreshold).toEqual(
+        marginThreshold
+      );
+    });
+  });
+
+  describe("onChange prop", () => {
```

Comment:
Maybe we can test that it doesn't break when onChange prop is not provided.

Reply:
^ what kind of break are you referring to? The snapshot test passes without this prop

Reply:
Basically to test the handleChange function, line 124 makes sure that if onChange prop isn't provided it won't call an undefined function.